src/IO/GeoJSONReader.php

```diff
@@ -0,0 +1,357 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\CoordinateSystem;
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Builds geometries out of GeoJSON Text strings.
+ */
+class GeoJSONReader
+{
+    /**
+     * @param string $geojson The GeoJSON to read.
+     *
+     * @return GeometryCollection|Geometry
+     * @throws \Brick\Geo\Exception\CoordinateSystemException
+     * @throws \Brick\Geo\Exception\GeometryIOException
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     * @throws \Brick\Geo\Exception\UnexpectedGeometryException
+     */
+    public function read(string $geojson) : Geometry
+    {
+        $geojsonArray = json_decode(strtoupper($geojson), true);
```

Comment:
I see this as well.... and I am happy to make this change... but do we really want to?  Not everyone follows this specification (as much as they should) and I am worried someone will have issues because of case sensitivity.  

The goal of this GeoJSONReader is to create Geometries from GeoJSON strings, not necessarily parse/validate the entire GeoJSON specification.  I don't think we should put more roadblocks than needed in the way of the end user.

What are your throughs @BenMorel ?

Reply:
I get your point, even though I'm not sure how common it is in other software to accept invalid GeoJSON files.

That being said, it doesn't hurt to be lenient with values (other than encouraging broken GeoJSON writer implementations!), however I would be strict on key case: GeoJSON is still JSON, and `"type": ...` is different from `"TYPE": ...` in every single JSON parser out there, so I would not make it possible to parse another key case.

Therefore, I would not `strtoupper()` the whole document, but only the specific value that we're comparing.